ec2.template

```diff
@@ -472,10 +605,24 @@
         ]
       }
     },
+    "AutoScalingGroupPublic": {
```

Comment:
AutoScalingGroupPublic / AutoScalingGroup outputs isn't ideal to me, but as discussed, CF doesn't let you do much in the output section, so if it's impossible to have conditionnal outputs with the same exact name, this is good enough backwards compatibility to me.